Remove mappings from the ServiceBinding resource (#154)
k8s-service-bindings/spec: README.md

A Provisioned Service resource **MUST** define a `.status.binding` which is a `LocalObjectReference`-able (containing a single field `name`) to a `Secret`.  The `Secret` **MUST** be in the same namespace as the resource.  The `Secret` data **SHOULD** contain a `type` entry with a value that identifies the abstract classification of the binding.  The `Secret` type (`.type` verses `.data.type`) **SHOULD** reflect this value as `service.binding/{type}`, replacing `{type}` with the `Secret` data type.  It is **RECOMMENDED** that the `Secret` data also contain a `provider` entry with a value that identifies the provider of the binding.  The `Secret` data **MAY** contain any other entry.  To facilitate discoverability, it is **RECOMMENDED** that a `CustomResourceDefinition` exposing a Provisioned Service add `service.binding/provisioned-service: "true"` as a label.

> > Note: While the Provisioned Service referenced `Secret` data should contain a `type` entry, the `type` must be defined before it is projected into an application workload. This allows a mapping to enrich an existing secret.

Reviewer: > a mapping to enrich

@scothis 
Are you referring to the "mappings" as described in #145?
Alternatively, the `.spec.type` (optional) value available in the `ServiceBinding` resource can be used in the binding Secret, right? Should we add that info into the note here?

Author: `.spec.type` is removed by this PR

Reviewer: I just noticed that now :-)

The removal of `.spec.type` and `.spec.provider` should be called out in the PR description because I don't see mentioning it in #145.

BTW, I am fine with removing `.spec.type` and `.spec.provider`. That way, the reconciler can mount the Secret resource into the application workload, and no need to create an intermediate Secret resource.

I hope soon the extensions mentioned in #145 will become a reality!

Author: > The removal of .spec.type and .spec.provider should be called out in the PR description

updated the PR subject and overview

> I hope soon the extensions mentioned in #145 will become a reality!

nothing to wait for. Start creating and using them today (they're just not part of the spec, yet)

> > Note: While the Provisioned Service referenced `Secret` data should contain a `type` entry, the `type` must be defined before it is projected into an application workload. This allows a mapping to enrich an existing secret.

Reviewer 2: ```suggestion
> Note: While the Provisioned Service referenced `Secret` data **SHOULD** contain a `type` entry, the `type` **MUST** be present in the application projection.
```

Author: this is a non-normative block, so it intentionally uses unstylized "should" and "must". The spec requirements are captured in the Application Projection section.
